tools/eksDistroBuildToolingOpsTools/pkg/issueManager/issueManager.go

```diff
@@ -89,4 +88,45 @@ func (p *IssueManager) CreateIssue(ctx context.Context, opts *CreateIssueOpts) (
 	logger.V(4).Info("sleeping after Issue creation to avoid secondary rate limiting by Github content API")
 	time.Sleep(time.Second * 1)
 	return issue, nil
-}
\ No newline at end of file
+}
+
+type GetIssueOpts struct {
+	Owner string
+	Repo  string
+	Issue int
+}
+
+func (p *IssueManager) GetIssue(ctx context.Context, opts *GetIssueOpts) (*gogithub.Issue, error) {
+	var issue *gogithub.Issue
+	var resp *gogithub.Response
+	var err error
+	issue, resp, err = p.client.Issues.Get(ctx, opts.Owner, opts.Repo, opts.Issue)
+	if resp != nil {
+		if resp.StatusCode == github.SecondaryRateLimitStatusCode {
+			b, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return nil, fmt.Errorf("reading Github response body: %v", err)
+			}
+			if strings.Contains(string(b), github.SecondaryRateLimitResponse) {
+				logger.V(4).Info("rate limited while attempting to get github issue")
+				return nil, fmt.Errorf("rate limited while attempting to get github issues: %v", err)
+			}
+		}
+
+		if resp.StatusCode == github.ResourceGoneStatusCode {
+			_, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return nil, fmt.Errorf("reading Github response body: %v", err)
+			}
+		}
+	}
+	if err != nil {
+		logger.V(4).Error(err, "getting Github issue", "response", resp)
+		return nil, fmt.Errorf("getting Github issue: %v; resp: %v", err, resp)
+	}
+	logger.V(4).Info("get issue response", "response", resp.Response.StatusCode)
+	logger.V(1).Info("Github issue received", "issue URL", issue.GetHTMLURL())
+	logger.V(4).Info("sleeping after Issue request to avoid secondary rate limiting by Github content API")
+	time.Sleep(time.Second * 1)
```

Comment:
I think for our purposes here we can remove these sleeps in the issue clients and add them in at a higher level if needed